src/crush/CrushCompiler.cc

```diff
@@ -270,7 +277,19 @@ int CrushCompiler::decompile(ostream &out)
 	break;
       case CRUSH_RULE_TAKE:
 	out << "\tstep take ";
-	print_item_name(out, crush.get_rule_arg1(i, j), crush);
+	{
+          int step_item = crush.get_rule_arg1(i, j);
+          int original_item;
+          int c;
+          int res = crush.split_id_class(step_item, &original_item, &c);
+          if (res < 0)
+            return res;
```

Comment:
if res < 0, doesn't that mean we didn't split, and this is a normal bucket with no class specified?